### PR TITLE
Prefer RAISE EXCEPTION NEW to RAISE EXCEPTION TYPE - new rule

### LINKED ISSUE #1607: Prefer RAISE EXCEPTION NEW to RAISE EXCEPTION TYPE
https://github.com/SAP/styleguides/blob/master/clean-abap/CleanABAP.md#prefer-raise-exception-new-to-raise-exception-type

### PR DESCRIPTION
**Pull request by jan-jezek**
Closes #1607

### CODE REVIEW

File: packages/core/src/rules/prefer_raise_exception_new.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+  public setConfig(conf: PreferRaiseExceptionNewConf) {
+    this.conf = conf;
+  }
+
+  public runParsed(file: ABAPFile): Issue[] {
+    const issues: Issue[] = [];
+
+    for (const statement of file.getStatements()) {
+      if (statement.concatTokens().toString().toUpperCase().startsWith("RAISE EXCEPTION TYPE ")) {
```
Comment (larshp) [inline packages/core/src/rules/prefer_raise_exception_new.ts:41]: concatenating all tokens is quite expensive suggest doing
`statement.get() instanceof Statements.Raise`

see example in https://github.com/abaplint/abaplint/blob/master/packages/core/src/rules/ambiguous_statement.ts

**Review by g-back — CHANGES_REQUESTED** [review]
Nice. Could you please add a version check? `RAISE EXCEPTION NEW` is only available from v752 onwards.
See https://github.com/abaplint/abaplint/blob/a86fd4892fb6c8fe12602dba45f4d28291c08095/packages/core/src/rules/inline_data_old_versions.ts#L35-L37 for an example.
Please also add test cases for different versions (above and below 752) - see the tests for the aforementioned rule for an example on how that works.

**Comment by jan-jezek** [comment]
Now with instanceof Statements.Raise this statement is not recognized.
`RAISE EXCEPTION TYPE cx_generation_error
  EXPORTING
    previous = exception
  MESSAGE e136(messages).`
because it is instanceof Unknown, created issue #1683.

**Comment by g-back** [comment]
Looks good, please add a test case where `RAISE EXCEPTION NEW` is used, then I'm happy 👍

**Review by g-back — APPROVED** [review]
thanks 👍

**Comment by larshp** [comment]
cool, merging

**Pull request merged**